File: 코딩 기초 트레이닝/수열과 구간 쿼리4-p181922/수열과 구간 쿼리4.py
# https://school.programmers.co.kr/learn/courses/30/lessons/181922

def solution(arr, queries):
    answer = []

    for q in queries:
        s = q[0]
        e = q[1]
        k = q[2]

        for index, a in enumerate(arr):
            if index >= s and index <= e and index % k == 0:
                arr[index] += 1

    return arr

File: 코딩 기초 트레이닝/수열과 구간 쿼리4-p181922/수열과 구간 쿼리4.js
// https://school.programmers.co.kr/learn/courses/30/lessons/181922

function solution(arr, queries) {
    const answer = [];

    queries.forEach((q) => {
        const s = q[0];
        const e = q[1];
        const k = q[2];

        arr = arr.map((a, index) => {
            if(index >= s && index <= e && index % k === 0 ) {
                return a +1;
            } else {
                return a;
            }
        });
    });

    return arr;
}
